Fix issue on auto discovery

f4iha opened the pull request:
Fixed the issue with the autodiscovery, the _service_matches_ is having a length of 0, however if we set the param _host_ from the discovered _addr_ and the _port_ at 1 it is fixing the problem

f4iha committed:
Fix issue on auto discovery

anfractuosity commented:
Thanks for your pull request!, will try and test this soon

anfractuosity commented:
Hi,

I've just added a couple more validation checks, to make sure that an address is obtained either from a parameter or
it's obtained by searching for it.

Unfortunately during my current testing, I can't seem to trigger the line:
            print("No services found for address ", addr, file=sys.stderr)

I'll try again, to see if I can get it to trigger, to see if your pull request fixes it for me.

Thanks

f4iha commented:
You will not see the line `No services found for address` appearing as I removed this part of the code and replaced by a static information.
Instead of having the error message and quit the code, I updated the code in order to use the same behavior as using the `--addr <ADDRESS>` with a specific ADDRESS defined
